_posts/2023-12-14-trusted-publishing.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Announcing Trusted Publishing on RubyGems.org
+layout: post
+author: Samuel Giddins
+author_email: segiddins@segiddins.me
+---
+
+Hi all!
+
+I'm excited to share a new feature that will help make RubyGems.org more secure, as well as making it easier to automate gem publishing. We're calling it [Trusted Publishing](https://guides.rubygems.org/trusted-publishing/).
+
+## Backstory
+
+Over the past few years, we've [increased the minimum multi-factor authentication (MFA) requirements](/2022/08/15/requiring-mfa-on-popular-gems) for accounts that own popular gems. We highly encourage requiring MFA for all interactions with RubyGems.org, including through the API. This is part of our strategy to make Ruby’s supply chain more secure, specifically by combatting account takeovers — the second most common software supply chain attack.
+
+On the flip side, we want to encourage gem authors to push their gems from trusted environments, based on the state of public repositories. For many, that means using GitHub Actions (as a CI/CD platform) for gems with repos hosted on GitHub.com.
+
+Until now, the desire to require MFA everywhere and to encourage automated gem pushing have been in opposition. Setting up automated publishing required disabling MFA on an API key that is stored as a long-lived secret on GitHub Actions. Even worse, given it was a clunky, multi-step process, many gem authors simply pushed gems from their laptops.
```

Comment:
```suggestion
Until now, MFA has been incompatible with automated gem pushes. You needed an API key with MFA disabled, stored by GitHub as a long-lived secret, and provided to every CI job. Even worse, given it was a clunky, multi-step process, many gem authors simply pushed gems from their laptops rather than go through the steps required to set that up.
```